packages/core/src/components/essentials/threshold.ts

```diff
@@ -86,7 +89,10 @@ export class Threshold extends Component {
 			.selectAll('g.threshold-group')
 			.data((d) =>
 				d.thresholds.map((threshold) => {
+					// Merge relevant keys into the threshold object
 					threshold.axisPosition = d.axisPosition;
+					threshold.correspondingDatasets = d?.correspondingDatasets;
+					threshold.mapsTo = d?.mapsTo;
 					return threshold;
```

Comment:
seems like `self.constructDatumObj(d)` should just run here right? edit the data right in the beginning instead of doing it multiple times